aws/solutions/ADConnector/src/adconnector_custom_resource.py

```diff
@@ -0,0 +1,134 @@
+"""Create and configure an ADConnector Directory.
+
+Copyright 2021 Amazon Web Services, Inc. or its affiliates. All Rights Reserved.
+This AWS Content is provided subject to the terms of the AWS Customer Agreement available at
+http://aws.amazon.com/agreement or other written agreement between Customer and either
+Amazon Web Services, Inc. or Amazon Web Services EMEA SARL or both.
+"""
+import json
+import logging
+import os
+
+import boto3
+from crhelper import CfnResource
+
+# Setup Default Logger
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+logger.setLevel(os.environ.get("LOG_LEVEL", logging.ERROR))
+
+# Initialize the helper
+helper = CfnResource(json_logging=False, log_level="DEBUG", boto_level="CRITICAL")
+
+try:
+    ds_client = boto3.client("ds")
+    secretsmanager_client = boto3.client("secretsmanager")
+except Exception as error:
+    helper.init_failure(error)
+
+
+def get_adconnector_parameters(params: dict) -> dict:
+    """Creates a parameters dictionary for the ds:connect_directory API call to create AD Connector.
+
+    Args:
+        params: event resource properties
+
+    Returns:
+        ADConnector Parameters
+    """
+    # Get AD Domain Join Credentials from Secrets Manager
+    response = secretsmanager_client.get_secret_value(SecretId=params["DOMAIN_JOIN_SECRET_ID"])
+    secret = json.loads(response["SecretString"])
+    # Create DNS Servers List
+
+    return {
+        "Name": params["DOMAIN_DNS_NAME"],
+        "ShortName": params["DOMAIN_NETBIOS_NAME"],
+        "Password": secret.get("password"),
+        "Description": params["ADCONNECTOR_DESCRIPTION"],
+        "Size": params["ADCONNECTOR_SIZE"],
+        "ConnectSettings": {
+            "VpcId": params["ADCONNECTOR_VPCID"],
+            "SubnetIds": [params["ADCONNECTOR_SUBNET_ID1"], params["ADCONNECTOR_SUBNET_ID2"]],
+            "CustomerDnsIps": params["DOMAIN_DNS_SERVERS"].replace(", ", ",").split(","),
+            "CustomerUserName": secret.get("username"),
+        },
+    }
+
+
+@helper.create
+def create(event, context) -> str:
+    """Create Event from AWS CloudFormation.
+
+    Args:
+        event: event data
+        context: runtime information
+
+    Raises:
+        Exception: Captures all exceptions.
+
+    Returns:
+        ADConnectorDirectoryResourceID
+    """
+    try:
+        logger.info("Create Event")
+        logger.info(f"REQUEST RECEIVED: {json.dumps(event, default=str)}")
+        adconnector_params = get_adconnector_parameters(event["ResourceProperties"])
+        response = ds_client.connect_directory(**adconnector_params)
+        logger.info(f"connect_directory_response = {json.dumps(response, default=str)}")
+        helper.PhysicalResourceId = response["DirectoryId"]
+        return helper.PhysicalResourceId
+    except Exception as error:
```

Comment:
This try/except block is redundant as immediately raising the error it captures is equivalent to not capturing it in the first place.

Reply:
(Also applies to the other methods in the file)